joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -18,6 +18,11 @@ trait AstCreatorHelper { this: AstCreator =>
    * */
   protected val methodAstParentStack: Stack[NewNode] = new Stack()
 
+  /* Use to if any constructors of classes are present to know if a default constructor should be generated
+   * TODO: this seems too specific to add another stack, perhaps there is a better way in checking the class body. There are some possible
```

Comment:
It may be possible to use one context-stack by defining a context-class used as a type for stack elements. Not sure if this makes sense here, but that's something that has worked well in the past.